open.go

```diff
@@ -186,6 +196,14 @@ func Open(connStr string) (*Dgraph, error) {
 			sslMode, sslModeDisable, sslModeRequire, sslModeVerifyCA)
 	}
 
+	if nsID != "" {
+		nsID, err := strconv.ParseUint(nsID, 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("invalid namespace ID: %w", err)
+		}
+		opts = append(opts, WithNamespace(nsID))
```

Comment:
If a namespace is specified, but a username and password are not, should this not return an error?